rholang/src/main/scala/coop/rchain/rholang/interpreter/PrettyPrinter.scala

```diff
@@ -174,16 +197,34 @@ case class PrettyPrinter(freeShift: Int, boundShift: Int) {
   private def buildPattern(patterns: Seq[Channel]): (Int, String) =
     ((0, "") /: patterns.zipWithIndex) {
       case ((patternsFree, string), (pattern, i)) =>
-        (patternsFree + freeCount(pattern), string + buildString(pattern) + {
-          if (i != patterns.length - 1) ", "
-          else ""
-        })
+        (patternsFree + freeCount(pattern),
+          string +
+            this
+              .copy(
+                freeId = boundId,
+                boundId = rotate(increment(baseId)),
+                baseId = increment(baseId)
+              )
+              .buildString(pattern) + {
+            if (i != patterns.length - 1) ", "
+            else ""
+          })
     }
 
   private def buildMatchCase(matchCase: MatchCase): String = {
     val patternFree: Int = matchCase.pattern.get.freeCount
-    PrettyPrinter(boundShift, 0).buildString(matchCase.pattern.get) + " => " +
-      PrettyPrinter(freeShift, boundShift + patternFree).buildString(matchCase.source.get)
+    "case " +
+      this
+        .copy(
+          freeShift = boundShift,
+          boundShift = 0,
+          freeId = boundId,
+          boundId = rotate(increment(boundId))
```

Comment:
This should be rotate(increment(baseId))
you should also increment basedId

Reply:
> This should be rotate(increment(baseId))
Thanks
> You should also increment basedId
I didn't increment baseId because cases are disjoint. It doesn't seem necessary to rotate the baseId if only one of them will be chosen.

Reply:
I don't think you understand.
boundId is already rotated.
You can't increment it accurately.